src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs

```diff
@@ -120,10 +120,20 @@ private async Task<Solution> ProcessResultAsync(CodeFixContext context, Diagnost
 
             // Now, rename all usages of the field to point at the property.  Except don't actually 
             // rename the field itself.  We want to be able to find it again post rename.
-            var updatedSolution = await Renamer.RenameAsync(fieldLocations, propertySymbol.Name,
-                location => !location.SourceSpan.IntersectsWith(declaratorLocation.SourceSpan) &&
-                            CanEditDocument(solution, location.SourceTree, linkedFiles, canEdit),
-                symbols => HasConflict(symbols, propertySymbol, compilation, cancellationToken),
+            //
+            // We're asking the rename API to update a bunch of references to an existing field to the same name as an
+            // existing property.  Rename will often flag this situation as an unresolvable conflict because the new
+            // name won't bind to the field anymore.
+            //
+            // To address this, we let rename know that there is no conflict if the new symbol it resolves to is the
+            // same as the property we're trying to get the references pointing to.
+
+            var updatedSolution = await Renamer.RenameAsync(
```

Comment:
FYI: I actually attempted to port this analyzer/code fix to shared layer, but it used an internal Rename API, so it did not get to complete it. It would be good that all APIs used here are public APIs, or at least we have a plan to make them public.

Reply:
yes. i will file issue on that.

Reply:
https://github.com/dotnet/roslyn/issues/43055